classes.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+class HistoryRepository
+{
+    protected $DS = DIRECTORY_SEPARATOR;
+    protected $savedData;
+
+    public function __construct($dirname)
+    {
+        $this->dirname = $dirname;
+
+        if(!is_dir($dirname)){
+            mkdir("{$this->dirname}");
+        }
+
+        fopen("{$this->dirname}{$this->DS}history.json", 'a+');
```

Comment:
let's move json file name to the protected field as well

```suggestion
        fopen("{$this->dirname}{$this->DS}{$this->fileName}", 'a+');
```